docs/data/routes/guides/code-patterns/layout-service/en.md

```diff
@@ -0,0 +1,93 @@
+---
+name: layout-service
+routeTemplate: ./data/component-templates/guide.yml
+title: Layout Service Customization Guide
+---
+
+> Tip: A reference of the Layout Service request model properties is documented in Sitecore Headless Development documentation - ["Sitecore Layout Service"](https://doc.sitecore.com/developers/100/developer-tools/en/sitecore-layout-service.html)
+
+---
+
+## When to customize Layout Service output
+
+JSS components are hydrated with dynamic data from the Layout Service output, which can be customized by developers to fulfill various business requirements. The OOTB techniques for augmenting Layout Service output are
+1. [Extending Sitecore Context](/docs/techniques/extending-layout-service/layoutservice-extending-context)
+2. [Using Rendering Contents Resolvers](/docs/techniques/extending-layout-service/layoutservice-rendering-contents#choose-or-configure-a-builtin-rendering-contents-resolver)
+3. [Using Integrated GraphQL](/docs/techniques/graphql/integrated-graphql)
+4. [Using Connected GraphQL](/docs/techniques/graphql/connected-graphql)
+
+### Security considerations
+
+Layout Service output is not secure, and can be inspected client-side. Treat it like an open API, ensuring you are not exposing sensitive or excess data.
+
+When extending `Sitecore Context` or building custom `Rendering Content Resolvers`:
+- Only add specific fields that are needed for front-end components. (Ex. Do not inject the entire Sitecore Context object from backend Sitecore pipelines into the Layout Service response)
+- Add data that represent global app state and needs to be accessed by multiple components
+- Avoid exposing sensitive information (Ex. Be aware of serialization depth to avoid accidentally exposing extra data via complex objects)
+
+---
+
+## Extending Sitecore Context
+
+Layout Service provides data about the `Sitecore Context` that a component is rendering in. On the front-end, `Sitecore Context` is available to all components. Extending `Sitecore Context` is great for data that represents global application state like `isUserLoggedIn` or `currentLanguage`.
+
+### Extending Sitecore Context in Disconnected Mode
+
+Community Guide
+- ["Adding Content Items to Sitecore JSS Disconnected Layout Service Context With customizeContext"](https://www.adamlamarre.com/adding-content-items-to-sitecore-jss-disconnected-layout-service-context-with-customizecontext/) by [Adam Lamarre](https://twitter.com/erzr)
+
```

Comment:
A few examples how to extend the GraphQL:
https://www.brimit.com/blog/graphql-extensibility-select-single-item-query
https://www.brimit.com/blog/graphql-extensibility-siterelativeitem-query